### PR TITLE
Small fixes from debugging

### PR DESCRIPTION
**Pull request by jxmorris12**
- Don't always set shuffle=True, set `shuffle=shuffle` in `dataloader_from_dataset` (that will always be set to true anyway so far, but this gives us the opportunity to set shuffle=False if we want to)
- Use multiple workers in Dataloaders (this should actually speed up training by a good amount)
- Add `model_weights_drop_linear` argument, which allows us to re-finetune a model that's already been finetuned once
- Change overloaded `batch` keyword in train.py to `train_batch` and `test_batch`

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -200,25 +216,25 @@ def run_training_loop(args: argparse.Namespace) -> str:
     training_step = 0
     for epoch in range(args.epochs):
         logger.info(f"Starting training epoch {epoch+1}/{args.epochs}")
-        for _epoch_step, batch in tqdm.tqdm(enumerate(train_dataloader), total=len(train_dataloader), desc='Training', leave=False):
+        for train_batch in tqdm.tqdm(train_dataloader, total=len(train_dataloader), desc='Training', leave=False):
```
Comment (jscuds) [inline train.py]: **Note** Small conflict with my PR #18 , I use the enumerated `_epoch_step` to save an example batch.

**Review by jscuds — COMMENTED** [review]
Looks great!  I made one comment, but it's an easy fix/rework.

**Comment by jxmorris12** [comment]
oh yeah let's merge #18 first and then one of us can fix the conflict by restoring the enumerate() on this branch.

File: train.py [hunk 2]
```diff
@@ -166,9 +168,23 @@ def run_training_loop(args: argparse.Namespace) -> str:
...
+            linear_keys = [k for k in model_weights.keys() if k.startswith('linear')]
+
+            if not len(linear_keys):
+                raise ValueError('--model_weights_drop_linear set but no weights for linear layers found')
+
+            for key in linear_keys:
+                logging.info(f'\t- dropping weight with key `%s`', key)
+                del model_weights[key]
+
+        missing_keys, unexpected_keys = experiment.model.load_state_dict(model_weights, strict=False)
```
Comment (jscuds) [inline train.py:193]: Clarification: in a model where we use `--model_weights_drop_linear` , the difference between training a new model, and re-tuning a finetuned model without its linear layers is...what?  The M-matrix would be different, right?  Is that the only difference?

**Review by jscuds — APPROVED** [review]
